tests/test_ensemble.py

```diff
@@ -0,0 +1,381 @@
+        np.testing.assert_allclose(rdf.table, np.array([[1,2],
+                                                        [2,9],
+                                                        [3,5]]))
+
+        #test invalid construction with r and g having different lengths
+        r = [1,2,3,4]
+        with self.assertRaises(ValueError):
+            rdf = relentless.RDF(r=r, g=g)
+
+        #test invalid construction with non-strictly-increasing r
+        r = [1,3,2]
+        with self.assertRaises(ValueError):
+            rdf = relentless.RDF(r=r, g=g)
+
+class test_Parallelepiped(unittest.TestCase):
+    """Unit tests for relentless.Parallelepiped"""
+
+    def test_init(self):
+        """Test creation from data."""
+        #test valid construction
+        p = relentless.Parallelepiped(a=(1,2,1),b=(3,4,5),c=(9,9,0))
+        np.testing.assert_allclose(p.a, np.array([1,2,1]))
+        np.testing.assert_allclose(p.b, np.array([3,4,5]))
+        np.testing.assert_allclose(p.c, np.array([9,9,0]))
+        np.testing.assert_allclose(p.matrix, np.array([[1,3,9],
+                                                       [2,4,9],
+                                                       [1,5,0]]))
+        self.assertAlmostEqual(p.volume, 36)
+
+        #test invalid construction
+        with self.assertRaises(TypeError):
+            p = relentless.Parallelepiped(a=(1,2,1),b=(3,4,5),c=(9,9))
+
+class test_Cuboid(unittest.TestCase):
+    """Unit tests for relentless.Cuboid"""
+
+    def test_init(self):
+        """Test creation from data."""
+        #test valid construction
+        c = relentless.Cuboid(Lx=3,Ly=4,Lz=5)
+        np.testing.assert_allclose(c.a, np.array([3,0,0]))
+        np.testing.assert_allclose(c.b, np.array([0,4,0]))
+        np.testing.assert_allclose(c.c, np.array([0,0,5]))
+        np.testing.assert_allclose(c.matrix, np.array([[3,0,0],
+                                                       [0,4,0],
+                                                       [0,0,5]]))
+        np.testing.assert_allclose(c.volume, 60)
+
+        #test invalid construction
+        with self.assertRaises(TypeError):
+            c = relentless.Cuboid(Lx=(3,4),Ly=4,Lz=5)
+
+class test_Cube(unittest.TestCase):
+    """Unit tests for relentless.Cube"""
+
+    def test_init(self):
+        """Test creation from data."""
+        #test valid construction
+        c = relentless.Cube(L=3)
+        np.testing.assert_allclose(c.a, np.array([3,0,0]))
+        np.testing.assert_allclose(c.b, np.array([0,3,0]))
+        np.testing.assert_allclose(c.c, np.array([0,0,3]))
+        np.testing.assert_allclose(c.matrix, np.array([[3,0,0],
+                                                       [0,3,0],
+                                                       [0,0,3]]))
+        np.testing.assert_allclose(c.volume, 27)
+
+        #test invalid construction
+        with self.assertRaises(TypeError):
+            c = relentless.Cube(L=(3,4,5))
+
```

Comment:
Add a test that an error is raised if `L <= 0`. (The Cuboid will do the error handling, so you don't need to modify `Cube`.)